server/src/parser.rs

```diff
@@ -497,12 +500,151 @@ fn column_defs_to_schema(column_defs: &Vec<ColumnDef>) -> Result<Schema, DataFus
     Ok(Schema::new(fields))
 }
 
+/* Start of code copied from datafusion-sql v14.0.0/v15.0.0. */
+// The following two functions have been copied from datafusion-sql v14.0.0/v15.0.0 as parser.rs
+// used the version of convert_simple_data_type() implemented as a free function in datafusion-sql
+// v14.0.0 to convert from sqlparser::ast::DataType (SQLDataType) to
+// datafusion::arrow::datatypes::DataType and it was changed to a private instance method in
+// datafusion-sql v15.0.0. As Apache Arrow DataFusion no longer seems to provide an API for
+// converting from SQLDataType to DataType, the version of convert_simple_data_type() in
+// datafusion-sql v14.0.0 has been copied to parser.rs and have been updated according to the
+// changes made in datafusion-sql v15.0.0 to the greatest degree possible. As the private function
+// make_decimal_type() is used by convert_simple_data_type() it has also been copied from
+// datafusion-sql v15.0.0. As these functions have been copied from datafusion-sql they should be
+// updated whenever a new version of datafusion-sql is released. Also significant effort should be
+// made to try and replace these two functions with calls to Apache Arrow DataFusion's public API
+// whenever a new version of Apache Arrow DataFusion is released.
+
+/// Convert a simple [`SQLDataType`] to the relational representation of the [`DataType`]. This
+/// function is copied from [datafusion-sql v14.0.0] and updated with the changes in [datafusion-sql
+/// v15.0.0] as it was changed from a public function to a private method in [datafusion-sql
+/// v15.0.0]. Both versions of datafusion-sql was released under Apache-2.0.
```

Comment:
Apache License 2.0?

+ "and are copyrighted by The Apache Software Foundation" ?